azure/container-apps/app_environment.tf

```diff
@@ -0,0 +1,37 @@
+resource "azurerm_container_app_environment" "app_environment" {
+  name                       = var.name
+  location                   = local.resource_group_location
+  resource_group_name        = local.resource_group_name
+  log_analytics_workspace_id = var.enable_log_analytics ? azurerm_log_analytics_workspace.log_analytics_workspace[0].id : null
+  dynamic "workload_profile" {
+    for_each = var.workload_profile != null ? [var.workload_profile] : []
+
+    content {
+      name                  = workload_profile.value.name
+      workload_profile_type = workload_profile.value.workload_profile_type
+    }
+  }
+}
+
+resource "azurerm_container_app_environment_certificate" "app_environment_certificate" {
+  count = var.dns_zone == null ? 0 : var.unique_environment_certificate != null ? 1 : length(var.container_apps)
```

Comment:
Can we do this without nested ternary operators?

Maybe something like this

```
locals {
  certificate_names = var.unique_environment_certificate != null ? 
    [var.unique_environment_certificate.name] : [for app in var.container_apps : "${app.name}-certificate"]
}

resource "azurerm_container_app_environment_certificate" "app_environment_certificate" {
  count = var.dns_zone == null ? 0 : length(local.certificate_names)

  name = local.certificate_names[count.index]

  ...
```

Reply:
great suggestion! :D